cms/static/js/views/import.js

```diff
@@ -2,241 +2,319 @@
+                completed: completed || false
+            }));
         };
 
         /**
-         * Check for import status updates every `timeout` milliseconds, and update
-         * the page accordingly.
-         * @param {string} url Url to call for status updates.
-         * @param {int} timeout Number of milliseconds to wait in between ajax calls
-         *     for new updates.
-         * @param {int} stage Starting stage.
+         * Sets the Import on the "success" status
+         *
+         * If it wasn't already, marks the stored import as "completed",
+         * and updates its date timestamp
          */
-        var getStatus = function (url, timeout, stage) {
-            var currentStage = stage || 0;
-            if (currentStage > 1) { CourseImport.okayToNavigateAway = true; }
-            if (CourseImport.stopGetStatus) { return ;}
-
-            if (currentStage === 4) {
-                // Succeeded
-                CourseImport.displayFinishedImport();
-                $('.view-import .choose-file-button').html(gettext("Choose new file")).show();
-            } else if (currentStage < 0) {
-                // Failed
-                var errMsg = gettext("Error importing course");
-                var failedStage = Math.abs(currentStage);
-                CourseImport.stageError(failedStage, errMsg);
-                $('.view-import .choose-file-button').html(gettext("Choose new file")).show();
-            } else {
-                // In progress
-                updateStage(currentStage);
+        var success = function () {
+            current.state = STATE.SUCCESS;
+
+            if (CourseImport.storedImport().completed !== true) {
+                storeImport(true);
             }
 
-            var time = timeout || 1000;
-            $.getJSON(url,
-                function (data) {
-                    setTimeout(function () {
-                        getStatus(url, time, data.ImportStatus);
-                    }, time);
-                }
-            );
+            destroyEventListeners();
+            updateFeedbackList();
+
+            deferred.resolve();
         };
 
+        /**
+         * Updates the Import feedback status list
+         *
+         * @param {string} [currStageMsg=''] The message to show on the
+         *   current stage (for now only in case of error)
+         */
+        var updateFeedbackList = function (currStageMsg) {
+            var $checkmark, $curr, $prev, $next;
+            var date, successUnix, time;
+
+            $checkmark = $dom.successStage.find('.icon');
+            currStageMsg = currStageMsg || '';
+
+            function completeStage(stage) {
+                $(stage)
+                    .removeClass("is-not-started is-started")
+                    .addClass("is-complete");
+            }
+
+            function errorStage(stage) {
+                var $stage = $(stage);
+                var error = currStageMsg;
```

Comment:
@AkA84 nit: this variables are used only once - might make sense to inline them.